### PR TITLE
508 for collections/editor/skeletalsubmit.php

### PR DESCRIPTION
**Pull request by NikitaSalikov**
# Before Changes :
Screenshot 2023-11-06 at 11 33 29 AM

# After Changes:

## Condensed Mode (Updated):
Screenshot 2023-11-10 at 10 33 43 AM

## Accessibility Mode (Updated):
Screenshot 2023-11-10 at 10 33 36 AM

# Changes Made:

- added Doctype and lang attribute
- fixed labels
- replaced fieldsets with fieldset-like sections
- removed "display instruction" functionality (duplicate of existing button in fieldset "Skeletal Data"
- added flex-form classes for accessibility mode. Removed extra inline stylings
- Commented out counter in the top right corner

# Comment for Ed:

I am not sure what goal had the counter in the top right corner so I commented out this section of code. If its functionality is no longer needed I can remove it entirely. Please advise on that. 

# Potential Issues:

- Worried about the layout of elements in fieldset. Not sure what should be the best. In particular, where to put "clear form" keeping in mind keyboard users. Any other directions will be welcomed!

# Pull Request Checklist:

# Pre-Approval

- [X] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [N/A] Hotfixes should be branched off of the `master` branch and **squash and merged** back into the `master` branch.
- [X] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [X] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages)
- [N/A] There are no linter errors
- [X] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [X] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [N/A] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [X] Comment which GitHub issue(s), if any does this PR address
- [N/A] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

### CODE REVIEW

**Review by themerekat — CHANGES_REQUESTED** [review]
@NikitaSalikov , in the Accessibility Mode, I think that the first two fields should still be side-by-side and followed by the green plus sign. The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?

_"I am not sure what goal had the counter in the top right corner so I commented out this section of code. If its functionality is no longer needed I can remove it entirely. Please advise on that."_

The counter only shows up once you have at least one record (maybe 2?) using this tool, so I assume you're just not seeing it yet. Its functionality **is** still needed.

_"Worried about the layout of elements in fieldset. Not sure what should be the best. In particular, where to put "clear form" keeping in mind keyboard users. Any other directions will be welcomed!"_

What if you put the "Clear Form" button at the very top of the fields? It's likely that if someone wants to clear the form, they do it before they start entering data, so it would logically make sense at the top.

File: collections/editor/skeletalsubmit.php [hunk 1]
```diff
@@ -109,15 +110,21 @@
 							<input name="addaction" type="radio" value="2" /> <?php echo $LANG['APPEND_VALUES']; ?>
 						</fieldset>
 					</div>
-					<div style="position:absolute;background-color:white;top:10px;right:10px;">
+					<!-- <div style="position:absolute;background-color:white;top:10px;right:10px;">
```
Comment (Atticus29) [inline collections/editor/skeletalsubmit.php]: If we need to comment something out, I'd prefer to delete it unless there's a really good reason to leave it in.

**Review by Atticus29 — COMMENTED** [review]
Looks good. 
TVP has some more feedback, though:
[tvp_collections_editor_skeletalsubmit.pdf](https://github.com/BioKIC/Symbiota/files/13342551/tvp_collections_editor_skeletalsubmit.pdf)

**Comment by Atticus29** [comment]
> The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?

Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?

Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.

Are these options alright?

**Comment by themerekat** [comment]
> > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> 
> Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> 
> Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> 
> Are these options alright?

Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?

**Comment by Atticus29** [comment]
> > > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> > 
> > 
> > Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> > Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> > Are these options alright?
> 
> Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?

@themerekat I was thinking either A:

Screen Shot 2023-12-11 at 12 50 29 PM

or B:

Screen Shot 2023-12-11 at 12 50 42 PM

Do you have a preference?

**Comment by Atticus29** [comment]
@NikitaSalikov, TVP still has some feedback for whenever a preference is expressed:

[tvp_collections_editor_skeletalsubmit.pdf](https://github.com/BioKIC/Symbiota/files/13640882/tvp_collections_editor_skeletalsubmit.pdf)

**Comment by themerekat** [comment]
> > > > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> > > 
> > > 
> > > Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> > > Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> > > Are these options alright?
> > 
> > 
> > Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?
> 
> @themerekat I was thinking either A:
> 
> Screen Shot 2023-12-11 at 12 50 29 PM
> or B:
> 
> Screen Shot 2023-12-11 at 12 50 42 PM
> Do you have a preference?

I like option B. This would only be enabled in Accessibility Mode, right?

**Comment by Atticus29** [comment]
> > > > > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> > > > 
> > > > 
> > > > Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> > > > Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> > > > Are these options alright?
> > > 
> > > 
> > > Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?
> > 
> > 
> > @themerekat I was thinking either A:
> > Screen Shot 2023-12-11 at 12 50 29 PM
> > or B:
> > Screen Shot 2023-12-11 at 12 50 42 PM
> > Do you have a preference?
> 
> I like option B. This would only be enabled in Accessibility Mode, right?

Accessibility mode would put them on separate lines. The separate labels for each input field is something that TotalValidatorPro throws an error, rather than a warning, about.

**Comment by Atticus29** [comment]
> > > > > > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> > > > > 
> > > > > 
> > > > > Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> > > > > Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> > > > > Are these options alright?
> > > > 
> > > > 
> > > > Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?
> > > 
> > > 
> > > @themerekat I was thinking either A:
> > > Screen Shot 2023-12-11 at 12 50 29 PM
> > > or B:
> > > Screen Shot 2023-12-11 at 12 50 42 PM
> > > Do you have a preference?
> > 
> > 
> > I like option B. This would only be enabled in Accessibility Mode, right?
> 
> Accessibility mode would put them on separate lines. The separate labels for each input field is something that TotalValidatorPro throws an error, rather than a warning, about.

There's also the option of hiding the Authorship label for anything except the screen readers, which might make it more visually pleasing and consistent with how this page has looked historically.

However, as a non-taxonomist getting used to Symbiota, I can say from personal experience that two input fields with the label of scientific name certainly tempt me to enter genus in field 1 and specific epithet in field 2. I think a little guidance even for our sighted, if less experienced taxonomist, end users isn't a horrible idea.

Furthermore, if we do go the route of hiding the label from sighted users and if said sighted users for whatever reason (e.g., dyslexia) prefer using accessibility mode, the authorship field would be both unlabeled and on its own line, which I think would definitely be non-ideal.

Thoughts?

**Comment by themerekat** [comment]
> > > > > > > The first two fields are scientific name and authorship, which are generally viewed as a single string. If the fields are side-by-side, do they not both read in through a screen reader?
> > > > > > 
> > > > > > 
> > > > > > Hi, @themerekat ! To a screen reader, the second field would just read as blank without the label. Ok if we give it the label, "Authorship"?
> > > > > > Another option is to put scientific name and authorship in a fieldset together, so that it's clear to a screen reader that they're grouped.
> > > > > > Are these options alright?
> > > > > 
> > > > > 
> > > > > Thanks for the explanation! Can you show me what it would look like to put the two in a fieldset together?
> > > > 
> > > > 
> > > > @themerekat I was thinking either A:
> > > > Screen Shot 2023-12-11 at 12 50 29 PM
> > > > or B:
> > > > Screen Shot 2023-12-11 at 12 50 42 PM
> > > > Do you have a preference?
> > > 
> > > 
> > > I like option B. This would only be enabled in Accessibility Mode, right?
> > 
> > 
> > Accessibility mode would put them on separate lines. The separate labels for each input field is something that TotalValidatorPro throws an error, rather than a warning, about.
> 
> There's also the option of hiding the Authorship label for anything except the screen readers, which might make it more visually pleasing and consistent with how this page has looked historically.
> 
> However, as a non-taxonomist getting used to Symbiota, I can say from personal experience that two input fields with the label of scientific name certainly tempt me to enter genus in field 1 and specific epithet in field 2. I think a little guidance even for our sighted, if less experienced taxonomist, end users isn't a horrible idea.
> 
> Furthermore, if we do go the route of hiding the label from sighted users and if said sighted users for whatever reason (e.g., dyslexia) prefer using accessibility mode, the authorship field would be both unlabeled and on its own line, which I think would definitely be non-ideal.
> 
> Thoughts?

The way that you interact with the field (by starting the enter the scientific name, and then a dropdown appears) would, I think, prevent people from mistaking the second field for specific epithet (it auto-fills after someone has selected a scientific name), but I definitely see your point. Also a valid point about the authorship being unlabeled when it's on its own line. I think it's ok to leave the label.

**Review by themerekat — APPROVED** [review]

**Review by Atticus29 — APPROVED** [review]
Made quite a few changes based on TVP, Katie's feedback, and my own discretion. Let's put off optimizing the keyboard navigation for the clear form button. Added it as a low-priority [rainy day ticket](https://github.com/orgs/BioKIC/projects/6/views/14?pane=issue&itemId=47357751).

**Pull request merged**